Feature/diplom

GlushkovV opened the pull request:
Выполнена дипломная работа.

GlushkovV committed:
Выполнено ДЗ 2

GlushkovV committed:
Исправлена ошибка "Объект типа Post должен создаваться при нажатии на кнопку и передаваться из FeedViewController в PostViewController, а не создаваться во viewDidLoad"

GlushkovV committed:
Выполнено ДЗ 3

GlushkovV committed:
Выполнено ДЗ 4

GlushkovV committed:
Выполнено задание 5.1

GlushkovV committed:
Выполнено ДЗ 5_2

GlushkovV committed:
UITableView, UIScrollView, datasource и delegate
Исправлено замечание:
Не хорошо, чтобы текст филд всегда видел в хедере. По заданию и по макетам
надо, чтобы при нажатии на кнопку Show Status он появлялся, а при нажатии
этой же кнопки на со стейтом Set Status, филд исчезал

GlushkovV committed:
Выполнено ДЗ 6

GlushkovV committed:
Выполнено ДЗ 6

GlushkovV committed:
Выполнено ДЗ 7

GlushkovV committed:
Добавлена авторизация, проверка корректности формата введенной почты,
проверка почты на колличество введенных символов, добавлен
лайбел с выводом ошибок, добавлен алерт с оповещением о корректном вводе учетных данных при не корректно веденных
данных но логине и пароле корректного формата с возможностью автозаполнения корректными тестовыми
данными.

 # Please enter the commit message for your changes. Lines
starting

GlushkovV committed:
Доработан экран с коллекцией фото, при нажатии анимировано открывается
картинка.

GlushkovV committed:
Исправлена локализация файлов проекта.

GlushkovV committed:
Исправлена перегрузка всех ViewController от излишних переменных для
констрейнтов.

GlushkovV committed:
Изменена реализация PhotosTableViewCell

GlushkovV committed:
Добавлено раскрытие аватара.

GlushkovV committed:
Добавлено удаление поста и анимация при пустом вводе статуса.

GlushkovV committed:
Изменена локализация файлов.

GlushkovV committed:
Добавлен просмотр постов с увеличением количества просмотров.

GlushkovV committed:
Реализованы лайки.

daddyUrsa commented:
Виталий, добрый день. 

**UI**
Логинка
![](https://u.netology.ngcdn.ru/backend/uploads/markdown_images/image/30355/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA_%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0_2022-04-08_%D0%B2_10.36.12.png)
Вот один из раскладов, когда жму на кнопку и ничего происходит. Желательно сделать так, чтобы при всех вариантах не верных данных было бы какие-то визуальные эффекты. 
Кстати очень понравилось решение с предложением заполнить логин и пароль.

Странная реализация: чтобы увидеть текстфилд нужно нажать на кнопку, нажимаю еще раз и она скрывается.Вы где-нибудь видели такое в приложениях? ))

Плохо смотрится такой пост. Очень много пробелов сверху и снизу
![](https://u.netology.ngcdn.ru/backend/uploads/markdown_images/image/30356/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA_%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0_2022-04-08_%D0%B2_10.40.43.png)

**Код**
Такие хранилища как `collectionDataSource: [CollectionViewModel]` надо убирать в отдельный файл. 

Убирайте закомментированный код. 

На этом все )) Исправляйте, присылайте ПР в телегу.

GlushkovV committed:
Добавлена анимация сообщений о ошибке в логинке. Скорректированы отсупы
в изображениях постов. Удален закомментированый код. Перемещено хранилище
collectionDataSource за пределы класса в папку моделей.

 # Please enter the commit
message for your changes. Lines starting

GlushkovV committed:
Изменена анимация установки статуса.
